tests/Unit/Model/Export/ProductTest.php

```diff
@@ -155,6 +155,83 @@ public function testEmptyAttributeGroups($productType)
         $this->assertEmpty($exportModel->getAttributeGroups());
     }
 
+    /**
+     * @param array     $imageData
+     * @param string    $smallImagePath
+     *
+     * @covers ::setImages
+     * @dataProvider imageProvider
+     */
+    public function testImageSortOrder($imageData, $smallImagePath)
+    {
+        $productStub = $this->getProductDouble();
+        $productStub->method('getMediaGalleryImages')
+            ->will($this->returnValue($this->getTestCollection($imageData)));
+
+        $productStub->method('getData')
+            ->will($this->returnValueMap([['small_image', null, $smallImagePath]]));
+
+        /** @var \Shopgate\Export\Model\Export\Product $exportModel */
+        $exportModel = $this->objectManager->getObject(
+            'Shopgate\Export\Model\Export\Product'
+        );
+
+        $exportModel->setItem($productStub)->setImages();
+
+        $images = array_reverse($exportModel->getImages());
+        $firstImage = array_pop($images);
+
+        $this->assertEquals($smallImagePath, $firstImage->getUrl());
+        $this->assertEquals(1, $firstImage->getIsCover());
+    }
+
+    /**
+     * @return array
+     */
+    public function imageProvider()
+    {
+        return [
+            'First product set as small image' => [[$this->createImageObject('1.jpg', '1'), $this->createImageObject('2.jpg', '3')], '1.jpg'],
+            'Second product set as small image' => [[$this->createImageObject('1.jpg', '1'), $this->createImageObject('2.jpg', '14')], '2.jpg']
+        ];
+    }
+
+    /**
+     * @param string    $url
+     * @param int       $position
+     *
+     * @return \Magento\Framework\DataObject
+     */
+    private function createImageObject($url, $position)
+    {
+        return new \Magento\Framework\DataObject([
+            'id'       => rand(0, 10000),
+            'url'      => $url,
+            'file'     => $url,
+            'position' => $position,
+            'tile'     => 'fake image',
+            'alt'      => 'fake image'
+        ]);
+    }
+
+    /**
+     * @param array $items
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject
```

Comment:
do we need to return fully qualified instead of psr0?

Reply:
Depends on the PHPUnit version. I was testing with an "old" not yet namespaced version.

Reply:
these module composers have no dependency on phpunit set up, so they use mage's, which is using v6.5+ as of 2.3

Reply:
Wrote the test for mage 2.1, should be phpunit 4.x!?

Reply:
ha, we kinda don't support 2.1 anymore :P

Reply:
you may need to add it as a composer dev dependency for lower phpunit if u want to go that route